examples/source/1.components/amp-script/index.html

```diff
@@ -282,6 +318,43 @@ <h2 id="translated-greeting"></h2>
         greeting.innerHTML = translation + '!';
       </script>
     </div>
+    <!--
+        ## Interacting with AMP components
+        Your script can communicate with an AMP component using a state variable.
+        Bind an attribute in the component to an expression containing a state variable.
+        When your script modifies that state variable, the change will propagate to the component.
+    -->
+    <div id="carousel-sample">
+      <amp-carousel
+        type="slides"
+        layout="responsive"
+        width="450"
+        height="300"
+        controls
+        loop
+        [slide]="slideIndex"
+        on="slideChange: AMP.setState({slideIndex: event.index})"
+      >
+        <amp-img src="/static/inline-examples/images/image1.jpg" layout="responsive" width="450" height="300"></amp-img>
+        <amp-img src="/static/inline-examples/images/image2.jpg" layout="responsive" width="450" height="300"></amp-img>
+        <amp-img src="/static/inline-examples/images/image3.jpg" layout="responsive" width="450" height="300"></amp-img>
+      </amp-carousel> 
+      <p>Slide <span [text]="slideIndex + 1">1</span> of 3</p>
+      
+      <amp-script layout="container" script="carousel-script" class="sample">
+        <button id="carousel-button">Surprise me</button>
+      </amp-script>
+
+      <script id="carousel-script" type="text/plain" target="amp-script">
+        const button = document.getElementById('carousel-button');
 
+        function gotoRandomSlide() {
```

Comment:
This also adds too much complexity. Simply switching between two images is enonugh.